src/main/java/edu/cornell/kfs/module/cam/document/service/impl/CuAssetServiceImpl.java

```diff
@@ -36,6 +40,21 @@
         return activeMatches;
     }
 
+    public Asset updateAssetInventory(String capitalAssetNumber, String conditionCode, String buildingCode, String roomNumber) {
+        Asset asset = businessObjectService.findBySinglePrimaryKey(Asset.class, capitalAssetNumber);
+        if (asset == null) {
```

Comment:
Since this is a business object being retrieved from the database, you may want to use ObjectUtils.isNotNull() to check for a non-null value, in case there's some kind of proxying going on.